project/JavaFormatter.scala

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2015 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.google.googlejavaformat.java.{Formatter, JavaFormatterOptions}
+import sbt.Keys._
+import sbt._
+import sbt.util.CacheImplicits._
+import sbt.util.{CacheStoreFactory, FileInfo, Logger}
+import scala.collection.immutable.Seq
+
+/** A local fork of https://github.com/sbt/sbt-java-formatter. The original plugin uses a very outdated version
+ * of `google-java-format` dependency which does not support syntax > JDK8. It's unlikely we will see the
+ * dependency upgraded any time soon, if ever, hence the almost one-to-one copy of the plugin.
+ */
```

Comment:
Shall we try to put a PR to the original one? Maybe they'd accept it?

(I'd suggest to do it in parallel to this change, we can keep our fork to have the benefits of the upgrade faster) But since you already did the work upgrading it anyway, I guess it may be welcomed if we can give it back to the community; what do you think?

Reply:
I think @hubertp mentioned that it is not worth trying to push it into the upstream. Right?

Reply:
As you can see it requires a bunch of extra options to jvm by default and a newer version of sbt. Both are likely going to take a while to be OK, if at all. I have a fork with the same changes so I can do a PR but I wouldn't have high hopes.